src/sinks/elasticsearch.rs

```diff
@@ -95,25 +263,36 @@ pub enum ElasticSearchAuth {
     Aws(AwsAuthentication),
 }
 
-#[derive(Derivative, Deserialize, Serialize, Clone, Debug)]
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub enum ElasticSearchMode {
+    Normal,
+    DataStream,
+}
+
+impl Default for ElasticSearchMode {
+    fn default() -> Self {
+        Self::Normal
+    }
+}
+
+#[derive(Derivative, Deserialize, Serialize, Clone, Copy, Debug)]
 #[serde(deny_unknown_fields, rename_all = "snake_case")]
-#[derivative(Default)]
 pub enum BulkAction {
-    #[derivative(Default)]
     Index,
     Create,
 }
 
 impl BulkAction {
-    pub fn as_str(&self) -> &'static str {
-        match *self {
+    pub fn as_str(self) -> &'static str {
```

Comment:
Usually `as_` methods take `&self`

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reply:
If I keep the `&self`, considering that `BulkAction` derives `Copy`, clippy starts complaining 🤔 
Initially I wanted to keep the `&self`, not sure if I should disable the clippy rule for that line or use `self`.

Reply:
Reverted this change and blocked this rule 😉